DocsClientYamlTestSuiteIT sometimes fails

`DocsClientYamlTestSuiteIT` sometimes fails because the order of the indices returned by the `_cat/indices` of the documentation snippet is wrong.

<details>
REPRODUCE WITH: gradle :docs:integTest -Dtests.seed=286B3ECD348881B0 -Dtests.class=org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT -Dtests.method="test {yaml=reference/cat/indices/line_8}" -Dtests.security.manager=true -Dtests.locale=und -Dtests.timezone=Iran

> [2016-10-21T15:24:37,056][INFO ][o.e.s.DocsClientYamlTestSuiteIT] Stash dump on failure [{
>   1>   "stash" : {
>   1>     "body" : "health status index    uuid                   pri rep docs.count docs.deleted store.size pri.store.size\ngreen  open   twitter2 _xA1Q6kuSSa6D6JvWLaaLg   5   0          0            0       260b           260b\nyellow open   twitter  -iylnuImQpahvry2rLVLBw   1   1       1200            0       130b           130b\n"
>   1>   }
>   1> }]
>   1> [2016-10-21T15:24:37,057][ERROR][o.e.s.DocsClientYamlTestSuiteIT] This failing test was generated by documentation starting at reference/cat/indices.asciidoc:line_8. It may include many snippets. See docs/README.asciidoc for an explanation of test generation.
>   2> REPRODUCE WITH: gradle :docs:integTest -Dtests.seed=286B3ECD348881B0 -Dtests.class=org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT -Dtests.method="test {yaml=reference/cat/indices/line_8}" -Dtests.security.manager=true -Dtests.locale=und -Dtests.timezone=Iran
> FAILURE 0.71s | DocsClientYamlTestSuiteIT.test {yaml=reference/cat/indices/line_8} <<< FAILURES!
> Throwable #1: java.lang.AssertionError: Failure at [reference/cat/indices:2456]: field [$body] was expected to match the provided regex but didn't
> Expected: health \s+status \s+index    \s+uuid                   \s+pri \s+rep \s+docs.count \s+docs.deleted \s+store.size \s+pri.store.size\s\* yellow \s+open   \s+twitter  \s+.+   \s+1   \s+1       \s+1200            \s+0     \s+\d+(.\d+)?[tgmk]?b         \s+\d+(.\d+)?[tgmk]?b\s\* green  \s+open   \s+twitter2 \s+.+   \s+5   \s+0          \s+0            \s+0       \s+\d+(.\d+)?[tgmk]?b           \s+\d+(.\d+)?[tgmk]?b\s*
>      but: was "health status index    uuid                   pri rep docs.count docs.deleted store.size pri.store.size\ngreen  open   twitter2 _xA1Q6kuSSa6D6JvWLaaLg   5   0          0            0       260b           260b\nyellow open   twitter  -iylnuImQpahvry2rLVLBw   1   1       1200            0       130b           130b\n"
>    at __randomizedtesting.SeedInfo.seed([286B3ECD348881B0:A03F01179A74EC48]:0)
>    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:329)
>    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:309)
>    at java.lang.Thread.run(Thread.java:745)
> Caused by: java.lang.AssertionError: field [$body] was expected to match the provided regex but didn't
> Expected: health \s+status \s+index    \s+uuid                   \s+pri \s+rep \s+docs.count \s+docs.deleted \s+store.size \s+pri.store.size\s\* yellow \s+open   \s+twitter  \s+.+   \s+1   \s+1       \s+1200            \s+0     \s+\d+(.\d+)?[tgmk]?b         \s+\d+(.\d+)?[tgmk]?b\s\* green  \s+open   \s+twitter2 \s+.+   \s+5   \s+0          \s+0            \s+0       \s+\d+(.\d+)?[tgmk]?b           \s+\d+(.\d+)?[tgmk]?b\s*
>      but: was "health status index    uuid                   pri rep docs.count docs.deleted store.size pri.store.size\ngreen  open   twitter2 _xA1Q6kuSSa6D6JvWLaaLg   5   0          0            0       260b           260b\nyellow open   twitter  -iylnuImQpahvry2rLVLBw   1   1       1200            0       130b           130b\n"
>    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
>    at org.elasticsearch.test.rest.yaml.section.MatchAssertion.doAssert(MatchAssertion.java:64)
>    at org.elasticsearch.test.rest.yaml.section.Assertion.execute(Assertion.java:76)
>    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:325)
>    ... 37 more
> IGNOR/A 0.01s | DocsClientYamlTestSuiteIT.test {yaml=reference/setup/cluster_restart/line_21}
> Assumption #1: [reference/setup/cluster_restart/line_21] skipped, reason: features [always_skip] not supported
> IGNOR/A 0.01s | DocsClientYamlTestSuiteIT.test {yaml=reference/indices/create-index/line_153}
> Assumption #1: [reference/indices/create-index/line_153] skipped, reason: features [always_skip] not supported
>   2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob-unix-compatibility/os/opensuse/docs/build/testrun/integTest/J0/temp/org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT_286B3ECD348881B0-001
>   2> NOTE: test params are: codec=Asserting(Lucene62), sim=RandomSimilarity(queryNorm=true,coord=crazy): {}, locale=und, timezone=Iran
>   2> NOTE: Linux 3.16.7-29-default amd64/Oracle Corporation 1.8.0_101 (64-bit)/cpus=4,threads=1,free=425831832,total=523239424
>   2> NOTE: All tests run in this JVM: [DocsClientYamlTestSuiteIT]
> Completed [1/1] in 115.11s, 665 tests, 1 failure, 15 skipped <<< FAILURES!
> 
> [ant:junit4] JVM J0: stderr was not empty, see: /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob-unix-compatibility/os/opensuse/docs/build/testrun/integTest/temp/junit4-J0-20161021_115306_814.syserr
> Tests with failures:
> - org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test {yaml=reference/cat/indices/line_8}

`twitter2` appears before `twitter` which is wrong.
</details>

--- a/docs/reference/cat/indices.asciidoc
+++ b/docs/reference/cat/indices.asciidoc
@@ -10,7 +10,12 @@ GET /_cat/indices/twi*?v&s=index
 --------------------------------------------------
 // CONSOLE
 // TEST[setup:huge_twitter]
+// TEST[s/^/POST _flush\n/]
 // TEST[s/^/PUT twitter2\n{"settings": {"number_of_replicas": 0}}\n/]
+// We flush very early here because the index's size is cached and we sort on
+// size below. So to get a realistic sort on size we need to flush here or else
+// the size is just whatever portion of the index is pushed out of memory
+// during test setup which isn't deterministic.
 
 Might respond with:
 
@@ -67,7 +72,6 @@ GET /_cat/indices?v&s=store.size:desc
 --------------------------------------------------
 // CONSOLE
 // TEST[continued]
-// TEST[s/^/POST _flush\n/]
 
 Which looks like:
 

